### PR TITLE
Per-request caching of DB queries

### PR DESCRIPTION
**Pull request by rafatower**
Doing some tests locally, this makes it go from 326  to 265 queries for the dashboard request. The cache is shared across requests but not between processes. This same approach can be applied to other frequent requests.

I checked the sequel cache plugin implementation and seems pretty decent in terms of ruby/sequel standards, so it makes little sense to implement my own stuff.

Another possiblity is to use a redis cache.

cc/ @javisantana @juanignaciosl @Kartones

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -10,7 +10,27 @@
 require_relative './external_data_import'
 require_relative './feature_flag'
 
+
```
Comment (rafatower) [inline app/models/user.rb]: obviously to be moved to a different file, if we go down this path.

**Comment by javisantana** [comment]
I don't like this approach because it assumes that user does not change across all the request (which can be right for most of the cases). I don't know if that plugin manages the invalidation too

In any case if this improves the performance...

**Comment by rafatower** [comment]
The plugin manages invalidation as far as the invalidation is done through the User class.

**Comment by Kartones** [comment]
A permission shouldn't change in the same request except upon creation (and similar with table). User id doesn't changes either... so sems relatively safe if invalidation is ok.

Tests will tell if permission and table creation work, but being temporal code until we remove Sequel I don't see it too bad.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2659/))

**Comment by rafatower** [comment]
@juanignaciosl the gem `request_store` works pretty well for this use-case (and after testing I'll be glad to pay for it)

I think this first in-memory cache level is pretty safe and will have a big gain in performance. We can think of a 2-level cache later on.

**Comment by juanignaciosl** [comment]
Great, I'll think about using it for the rewrite branch as well :-)

**Comment by rafatower** [comment]
retest this, please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2705/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2707/))

**Comment by rafatower** [comment]
retest this, please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2718/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2764/))

**Pull request merged**